File: app/src/main/java/de/xikolo/controllers/dialogs/ProgressDialog.java
package de.xikolo.controllers.dialogs;

import android.app.Dialog;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.DialogFragment;

import de.xikolo.App;
import de.xikolo.R;

public class ProgressDialog extends DialogFragment {

    public static final String TAG = ProgressDialog.class.getSimpleName();

    public static final String ARG_TITLE = "multiple_title";
    public static final String ARG_MESSAGE = "multiple_message";

    private String title;

    private String message;

    public ProgressDialog() {
    }

    public static ProgressDialog getInstance() {
        return getInstance(App.getInstance().getString(R.string.dialog_progress_title),
                App.getInstance().getString(R.string.dialog_progress_message));
    }

    public static ProgressDialog getInstance(String title, String message) {
        ProgressDialog fragment = new ProgressDialog();
        Bundle args = new Bundle();
        args.putString(ARG_TITLE, title);
        args.putString(ARG_MESSAGE, message);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    @NonNull
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        if (getArguments() != null) {
            title = getArguments().getString(ARG_TITLE);
            message = getArguments().getString(ARG_MESSAGE);
        }

        android.app.ProgressDialog dialog = new android.app.ProgressDialog(getActivity());
        dialog.setTitle(title);
        dialog.setMessage(message);
        dialog.setCancelable(false);
        dialog.setCanceledOnTouchOutside(false);

        setCancelable(false);

        return dialog;
    }

}

File: app/src/main/java/de/xikolo/controllers/dialogs/ProgressDialog.kt
package de.xikolo.controllers.dialogs

import android.app.Dialog
import android.os.Bundle
import android.support.v7.app.AlertDialog
import android.widget.ProgressBar
import com.yatatsu.autobundle.AutoBundleField
import de.xikolo.App
import de.xikolo.R
import de.xikolo.controllers.dialogs.base.BaseDialogFragment

class ProgressDialog : BaseDialogFragment() {

    companion object {
        @JvmField
        val TAG: String = ProgressDialog::class.java.simpleName
    }

    @AutoBundleField(required = false)
    var title: String? = null

    @AutoBundleField(required = false)
    var message: String? = null

    var max = 0

    var progressStyle = android.app.ProgressDialog.STYLE_HORIZONTAL

    var progress = 0

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val dialog = android.app.ProgressDialog(activity)
        dialog.setTitle(title)
        dialog.setMessage(message)
        dialog.setCancelable(false)
        dialog.setCanceledOnTouchOutside(false)

        isCancelable = false

        val p: ProgressBar = layoutInflater.inflate(R.layout.container_progress_bar, null) as ProgressBar

        val b = AlertDialog.Builder(App.getInstance())
            .setTitle(title)
            .setMessage(message)
            .setCancelable(false)
            .setView(p)

        return dialog
    }

}
